File: crawl_module/domain/hiring_post.py
from datetime import datetime

from peewee import *

from crawl_module.domain.base_model import BaseModel
from crawl_module.domain.company import TbCompany


class TbHiringPost(BaseModel):
    post_id = BigAutoField()
    origin_post_id = BigIntegerField(null=True)
    company_id = ForeignKeyField(TbCompany, backref="hiring_posts")
    job_category = CharField(max_length=255, null=True)
    post_title = CharField(max_length=255)
    content = TextField(null=True)
    recruitment_process = TextField(null=True)
    min_career = CharField(max_length=255, null=True)
    max_career = CharField(max_length=255, null=True)
    education = CharField(max_length=255, null=True)
    employment_type = CharField(max_length=255, null=True)
    working_region = CharField(max_length=255, null=True)
    coding_test_exist_yn = CharField(max_length=1, default="N")
    assignment_exist_yn = CharField(max_length=1, default="N")
    only_image_yn = CharField(max_length=1, default="N")
    admin_checked_yn = CharField(max_length=1, default="N")
    post_url = CharField(max_length=255)
    created_at = DateTimeField(default=datetime.now)
    updated_at = DateTimeField(default=datetime.now)

    class Meta:
        db_table = "TB_HIRING_POST"

File: src/main/java/com/why_not_cote/entity/post/HiringPost.java
package com.why_not_cote.entity.post;

import com.why_not_cote.entity.CommonBaseDateTime;
import com.why_not_cote.entity.company.Company;
import com.why_not_cote.util.code.YnCode;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "TB_HIRING_POST")
public class HiringPost extends CommonBaseDateTime {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "post_id")
    private Long postId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id")
    private Company company;

    @Column(name = "origin_post_id")
    private Long originPostId;

    @Column(name = "job_category")
    private String jobCategory;

    @Column(name = "post_title", nullable = false)
    private String postTitle;

    @Column(name = "content")
    private String content;

    @Column(name = "recruitment_process")
    private String recruitmentProcess;

    @OneToMany(targetEntity = Skill.class)
    @JoinTable(name = "TB_POST_SKILL",
            joinColumns = @JoinColumn(name = "POST_ID"),
            inverseJoinColumns = @JoinColumn(name = "SKILL_ID"))
    private List<Skill> SkillList;

    @Column(name = "min_career")
    private String minCareer;

    @Column(name = "max_career")
    private String maxCareer;

    @Column(name = "education")
    private String education;

    @Column(name = "employment_type")
    private String employmentType;

    @Column(name = "working_region")
    private String workingRegion;

    @Column(name = "coding_test_exist_yn")
    @Enumerated(EnumType.STRING)
    private YnCode codingTestYn;

    @Column(name = "assignment_exist_yn")
    @Enumerated(EnumType.STRING)
    private YnCode assignmentYn;

    @Column(name = "only_image_yn")
    @Enumerated(EnumType.STRING)
    private YnCode onlyImageYn;

    @Column(name = "admin_checked_yn")
    @Enumerated(EnumType.STRING)
    private YnCode adminCheckedYn;

    @Column(name = "post_url")
    private String postUrl;

}
